src/main/java/nl/dtls/fairdatapoint/service/metadata/common/AbstractMetadataService.java

```diff
@@ -186,7 +189,12 @@ public void delete(IRI uri, ResourceDefinition rd) throws MetadataServiceExcepti
     protected void updateParent(Model metadata, IRI uri, ResourceDefinition rd) throws MetadataServiceException {
         IRI parent = MetadataGetter.getParent(metadata);
         if (parent != null) {
-            ResourceDefinition rdParent = resourceDefinitionCache.getParentByUuid(rd.getUuid());
+            String[] parts = parent.toString().split("/");
+            if (parts.length < 3) {
+                throw new ValidationException("Invalid parent uri");
+            }
+            String parentPrefix = parts[parts.length-2];
```

Comment:
The string splitting seems fragile, but assuming the values are always absolute `IRI`s it seems to be a reasonable way of handling it.

Reply:
Yes, it always gets absolute IRIs... this could be refactored (as it is on more places in the code) together with RD URI prefix validations and handling.